File: MyMindNotes/app/src/main/java/com/android/mymindnotes/model/retrofit/UpdateDiaryApi.java
package com.android.mymindnotes.model.retrofit;

import com.android.mymindnotes.data.retrofit.model.diary.DiaryEdit;

import java.util.Map;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.PUT;
import retrofit2.http.Path;

public interface UpdateDiaryApi {
    @PUT("/api/diary/update/{diary_number}")
    Call<Map<String, Object>> updateDiary(@Path("diary_number") int diary_number, @Body DiaryEdit diaryEdit);
}

File: MyMindNotes/data/src/main/java/com/android/mymindnotes/data/retrofit/api/diary/UpdateDiaryApi.kt
package com.android.mymindnotes.data.retrofit.api.diary

import com.android.mymindnotes.core.dto.UpdateDiaryResponse
import retrofit2.http.PUT
import com.android.mymindnotes.core.dto.DiaryEdit
import retrofit2.http.Body
import retrofit2.http.Path

interface UpdateDiaryApi {
    @PUT("/api/diary/update/{diary_number}")
    suspend fun updateDiary(
        @Path("diary_number") diary_number: Int,
        @Body diaryEdit: DiaryEdit
    ): UpdateDiaryResponse
}